docs/DeveloperGuide.md

```diff
@@ -1,38 +1,662 @@
+    - `CustomExceptions.InsufficientInput` is thrown if either not enough parameters are specified or blank parameters are found.
+    - `CustomExceptions.InvalidInput` is thrown if the parameters passed in are invalid and do not follow the stipulated format.
 
-### Value proposition
+4. If valid, a new `Run` object is created with the split user input.
 
-{Describe the value proposition: what problem does it solve?}
+5. The `Run` constructor adds the newly created object into `WorkoutList.WORKOUTS` and `WorkoutList.RUNS`.
 
-## User Stories
+6. The `Run` object is passed to `Output.printAddRun()` and a message acknowledging the successful adding is printed to the screen.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+This is the sequence diagram for adding a run:
 
-## Non-Functional Requirements
+![Run Sequence Diagram](img/run_sequence_diagram.png)
 
-{Give non-functional requirements}
+###### [Back to table of contents](#table-of-contents)
 
-## Glossary
+#### Add Gym
 
-* *glossary item* - Definition
+<code style="color: #D85D43;">
+WORKOUT /e:gym /n:[number_of_stations] /date:[date]
+</code>
 
-## Instructions for manual testing
+- `[number_of_stations]` is a **positive integer** representing the number of gym stations done.
+- `[date]` is in `DD-MM-YYYY` format. **THIS IS AN OPTIONAL PARAMETER**.
+
+The bot will then prompt the user for the specific gym details in this format:
+
+<code style="color: #D85D43;">
+[station_name] /s:[number_of_sets] /r:[number_of_repetitions] /w:[weight]
+</code>
+
+- `[station_name]` is a string containing **only alphanumeric characters and spaces**, representing the station name.
+- `[number_of_sets]` is a **positive integer** representing the number of sets done for the current station.
+- `[number_of_repetitions]` is a **positive integer** representing the number of repetitions done for the current station.
+- `[weight]` is a string containing **positive integers separated by commas**, representing the weight done for each set in kilograms.
+    - For example, if the user has done 3 sets of 10kg, 20kg and 15kg, the string entered is `10,20,15`.
+    - **No spaces can be present in this string**.
+
+Example of Gym Station:
+
+<code style="color: #D85D43;">Bench Press /s:2 /r:3 /w:100,200 </code>
+
+- Station name is `Bench Press`.
+- User has done **2 sets of 3 repetitions**.
+- The first set was done using 100kg, and the second using 200kg.
+
+> NOTE: The number of sets entered matches the number of weights added. If 4 sets have been done, the bot expects 4 positive inters separated by commas as the `weights` input.
+
+##### Gym Sequence
+
+![Gym Sequence Diagram](./img/gym_sequence_diagram.jpg)
```

Comment:
![Screenshot 2024-04-04 123049](https://github.com/nus-cs2113-AY2324S2/tp/assets/110781598/1160dcc1-c214-45ab-ab09-64fae499b480)
Is there inconsistency in naming? The first four says <<class>> and has colon before the name, while the last says <<Class>> and has no colon, resulting in some confusion